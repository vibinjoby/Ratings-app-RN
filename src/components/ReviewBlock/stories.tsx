import React from 'react'
import { View, StyleSheet } from 'react-native'
import { storiesOf } from '@storybook/react-native'

import ReviewBlock from './component'

storiesOf('Components', module).add('ReviewBlock', () => (
  <View style={styles.container}>
    <ReviewBlock
      reviewId="123"
      reviewerName="Vibin"
      comments="This is not too bad at all"
      ratings={3}
      visitDate="20 Jun 2021"
      shouldReply
    />
  </View>
))

const styles = StyleSheet.create({
  container: {
    padding: 20,
  },
})
