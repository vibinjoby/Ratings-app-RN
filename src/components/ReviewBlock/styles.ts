import { StyleSheet } from 'react-native'
import Typography from '../../utilities/typography'

export default StyleSheet.create({
  container: {
    flex: 1,
  },
  reviewContainer: {
    flexDirection: 'row',
    marginTop: 20,
  },
  reviewWrapper: {
    marginLeft: 20,
    flex: 1,
  },
  reviewerName: {
    ...Typography.Body.bold,
    fontSize: 16,
  },
  ratingWrapper: {
    flexDirection: 'row',
  },
  comments: {
    ...Typography.RegularFont,
    fontSize: 18,
    marginVertical: 2,
  },
  visitDt: {
    ...Typography.Caption1.regular,
    fontSize: 10,
    marginLeft: 10,
  },
  replyText: {
    alignSelf: 'flex-start',
    ...Typography.Caption1.regular,
    fontSize: 12,
  },
})
