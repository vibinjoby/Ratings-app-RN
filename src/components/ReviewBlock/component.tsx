import React from 'react'
import { useState } from 'react'
import { Text, View, Image } from 'react-native'

import Colors from '../../utilities/colors'
import OwnerReply from '../OwnerReply'
import ReplyReview from '../ReplyReview'
import Stars from '../Stars'
import styles from './styles'

export interface ReviewBlockProps {
  reviewId: string
  reviewerName: string
  comments: string
  ratings: number
  visitDate: string /* eslint-disable camelcase */
  owner_reply?: string
  shouldReply?: boolean
  onSend?: (arg0: string, arg1: string) => void
}

const ReviewBlock: React.FC<ReviewBlockProps> = ({
  reviewId,
  reviewerName,
  comments,
  ratings,
  visitDate,
  owner_reply,
  shouldReply,
  onSend,
}: ReviewBlockProps) => {
  const [isReplyPressed, setIsReplyPressed] = useState(false)
  const ReplyButton = () => (
    <Text style={styles.replyText} onPress={() => setIsReplyPressed(true)}>
      Reply
    </Text>
  )

  const onReply = (reviewId: string, replyComments: string) => {
    setIsReplyPressed(false)
    onSend && onSend(reviewId, replyComments)
  }
  return (
    <View style={styles.reviewContainer}>
      <Image source={require('../../assets/dummyUser/dummyUser.png')} />
      <View style={styles.reviewWrapper}>
        <Text style={styles.reviewerName}>{reviewerName?.toUpperCase()}</Text>
        <View style={styles.ratingWrapper}>
          <Stars
            selectedStars={ratings}
            selectable={false}
            starHeight={14}
            selectedColor={Colors.appOrange}
          />
          <Text style={styles.visitDt}>{visitDate.substr(0, 13)}</Text>
        </View>
        <Text style={styles.comments}>{comments}</Text>
        {isReplyPressed && <ReplyReview _id={reviewId} onSend={onReply} />}
        {shouldReply && !owner_reply && !isReplyPressed && <ReplyButton />}
        {/* eslint-disable camelcase */}
        {owner_reply ? <OwnerReply owner_reply={owner_reply} /> : <></>}
      </View>
    </View>
  )
}

export default ReviewBlock
