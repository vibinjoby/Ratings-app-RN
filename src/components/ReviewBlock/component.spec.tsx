import 'react-native'
import React from 'react'
import { render } from '@testing-library/react-native'

import ReviewBlock from './component'

describe('ReviewBlock', () => {
  it('should render', () => {
    const { toJSON } = render(
      <ReviewBlock
        reviewId="123"
        reviewerName="Vibin"
        comments="This is not too bad at all"
        ratings={3}
        visitDate="20 Jun 2021"
      />,
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
