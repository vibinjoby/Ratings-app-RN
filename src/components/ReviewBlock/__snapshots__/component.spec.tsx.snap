// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewBlock should render 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginTop": 20,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/dummyUser/dummyUser.png",
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#181725",
          "fontFamily": "Poppins-Bold",
          "fontSize": 16,
        }
      }
    >
      VIBIN
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <RCTScrollView
        ItemSeparatorComponent={[Function]}
        data={
          Array [
            1,
            2,
            3,
            4,
            5,
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEnabled={false}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        viewabilityConfigCallbackPairs={Array []}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#FF5823",
                      "fontSize": 14,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "marginRight": 5,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#FF5823",
                      "fontSize": 14,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "marginRight": 5,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#FF5823",
                      "fontSize": 14,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "marginRight": 5,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#A5A6B7",
                      "fontSize": 14,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "marginRight": 5,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#A5A6B7",
                      "fontSize": 14,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <Text
        style={
          Object {
            "color": "#7C7C7C",
            "fontFamily": "Poppins-Regular",
            "fontSize": 10,
            "marginLeft": 10,
          }
        }
      >
        20 Jun 2021
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontFamily": "Poppins-Regular",
          "fontSize": 18,
          "marginVertical": 2,
        }
      }
    >
      This is not too bad at all
    </Text>
  </View>
</View>
`;
